bootstrap.sh

```diff
@@ -61,6 +65,55 @@ function self_update_if_its_time() {
     fi
 }
 
+function fail {
+  echo $1 >&2
+  exit 1
+}
+
+function retry {
+  local n=1
+  local max=10
+  local delay=3
```

Comment:
We have pretty constraints at the top, what do you think of moving these two bad boys there too?